### PR TITLE
consensus/clique: prevent 0 len extradata from panicing

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
closes https://github.com/ethereum/go-ethereum/issues/23522

### CODE REVIEW

**Comment by karalabe** [comment]
This was a known bug that we deliberately didn't fix. Perhaps just crashing isn't ideal, but a network with 0 signers isn't meaningful either. Perhaps we should add a genesis validation vs. just silently make a non functional chain.

**Comment by MariusVanDerWijden** [comment]
Moved the check to `genesis.Commit()`

**Comment by karalabe** [comment]
Btw, I think with the genesis check, you can still shoot yourself in the foot if the signer outvotes itself. But :D, not sure we want to do all the idiot proofing :)

File: core/genesis_test.go [hunk 1]
```diff
@@ -39,6 +39,22 @@ func TestDefaultGenesisBlock(t *testing.T) {
 	if block.Hash() != params.RopstenGenesisHash {
 		t.Errorf("wrong ropsten genesis hash, got %v, want %v", block.Hash(), params.RopstenGenesisHash)
 	}
+	block = DefaultRinkebyGenesisBlock().ToBlock(nil)
+	if block.Hash() != params.RinkebyGenesisHash {
+		t.Errorf("wrong ropsten genesis hash, got %v, want %v", block.Hash(), params.RinkebyGenesisHash)
```
Comment (holiman) [inline core/genesis_test.go:44]:
```suggestion
		t.Errorf("wrong rinkeby genesis hash, got %v, want %v", block.Hash(), params.RinkebyGenesisHash)
```

File: core/genesis_test.go [hunk 2]
```diff
@@ -39,6 +39,22 @@ func TestDefaultGenesisBlock(t *testing.T) {
...
+	}
+	block = DefaultGoerliGenesisBlock().ToBlock(nil)
+	if block.Hash() != params.GoerliGenesisHash {
+		t.Errorf("wrong ropsten genesis hash, got %v, want %v", block.Hash(), params.GoerliGenesisHash)
```
Comment (holiman) [inline core/genesis_test.go:48]:
```suggestion
		t.Errorf("wrong goerli genesis hash, got %v, want %v", block.Hash(), params.GoerliGenesisHash)
```

**Review by holiman — APPROVED** [review]

**Pull request merged**